File: .agents/skills/glimpse-design-system/scripts/language.py
"""Load review text in the user's language."""
import json
import re
from pathlib import Path

ASSETS = Path(__file__).resolve().parents[1] / 'assets'


def translated_copy(language, overrides=None, asset='studio-copy.json'):
    if not isinstance(language, str) or not re.fullmatch(r'[A-Za-z]{2,8}(?:-[A-Za-z0-9]{1,8})*', language):
        raise ValueError('Provide a language tag such as zh-CN, en or ja')
    copies = json.loads((ASSETS / asset).read_text())
    base = 'en' if language.lower().split('-')[0] == 'en' else None
    result = dict(copies[base]) if base else {}
    if overrides is not None:
        if not isinstance(overrides, dict):
            raise ValueError('UI copy must be an object of translated strings')
        result.update(overrides)
    required = copies['en']
    if any(not isinstance(result.get(key), str) or not result[key].strip() for key in required):
        raise ValueError('Translate every UI text key for ' + language + ' using --ui-copy')
    return result

File: skills/grilling-design-system/scripts/language.js
import fs from 'node:fs';
import path from 'node:path';
import { fileURLToPath } from 'node:url';

export const ASSETS = path.resolve(path.dirname(fileURLToPath(import.meta.url)), '..', 'assets');

/** Load review text in the user's language. */
export function translated_copy(language, overrides = undefined, asset = 'studio-copy.json') {
  if (typeof language !== 'string' || !/^[A-Za-z]{2,8}(?:-[A-Za-z0-9]{1,8})*$/.test(language)) {
    throw new Error('Provide a language tag such as zh-CN, en or ja');
  }
  const copies = JSON.parse(fs.readFileSync(path.join(ASSETS, asset), 'utf8'));
  const base = language.toLowerCase().split('-')[0] === 'en' ? 'en' : null;
  const result = base ? { ...copies[base] } : {};
  if (overrides !== undefined && overrides !== null) {
    if (typeof overrides !== 'object' || Array.isArray(overrides)) {
      throw new Error('UI copy must be an object of translated strings');
    }
    Object.assign(result, overrides);
  }
  if (Object.keys(copies.en).some((key) => typeof result[key] !== 'string' || !result[key].trim())) {
    throw new Error(`Translate every UI text key for ${language} using --ui-copy`);
  }
  return result;
}
